### ISSUE TITLE
buffalo build fails in strange condition (v0.18.0)

### ISSUE THREAD

**Issue by sio4** [gobuffalo/cli#62]
### Description

`buffalo build` works fine for the first time, but after that, it fails with the following error message:

```console
$ buffalo build -v -o zzz
DEBU[2021-11-29T00:22:20+09:00] Step: dd26fa27
DEBU[2021-11-29T00:22:20+09:00] Chdir: /home/sio4/git/bt/app
DEBU[2021-11-29T00:22:20+09:00] Exec: go mod tidy
2021/11/29 00:22:20 build:clean fork/exec /opt/google/go/bin/go: too many open files
$ 
```

If I manually remove `bin/app`, it works fine again. However, the issue is not a previously built app. If I copied e.g. `buffalo` binary into the app root, `buffalo build` does not work. After some try, I found the exact condition:

- any C binary such as `bash`, `wireshark` doesn't affect
- if I drop any binary built with Go within the app directory with a file name after `node_modules`, e.g. `node_moduless`, it does not affect.
- if I drop the same binary within the app directory with a filename before `node_modules`, e.g. `node_module`, `buffalo build` will not work.

### Steps to Reproduce the Problem

1. go install -v -tags sqlite github.com/gobuffalo/cli/cmd/buffalo@v0.18.0
2. buffalo new app
3. cd app
4. buffalo build -o zzz
5. buffalo build
6. buffalo build

### Info

<details>

```
-> Go: Checking installation
✓ The `go` executable was found on your system at: /opt/google/go/bin/go

-> Go: Checking minimum version requirements
✓ Your version of Go, 1.16.7, meets the minimum requirements.

-> Go: Checking Package Management
✓ You are using Go Modules (`go`) for package management.

-> Go: Checking PATH
✓ Your PATH contains /home/sio4/go/bin.

-> Node: Checking installation
✓ The `node` executable was found on your system at: /home/sio4/.nvm/versions/node/v14.17.6/bin/node

-> Node: Checking minimum version requirements
✓ Your version of Node, v14.17.6, meets the minimum requirements.

-> NPM: Checking installation
✓ The `npm` executable was found on your system at: /home/sio4/.nvm/versions/node/v14.17.6/bin/npm

-> NPM: Checking minimum version requirements
✓ Your version of NPM, 6.14.15, meets the minimum requirements.

-> Yarn: Checking installation
✓ The `yarnpkg` executable was found on your system at: /home/sio4/.nvm/versions/node/v14.17.6/bin/yarnpkg

-> Yarn: Checking minimum version requirements
✓ Your version of Yarn, 1.22.11, meets the minimum requirements.

-> PostgreSQL: Checking installation
✘ The `postgres` executable could not be found on your system.
For help setting up your Postgres environment please follow the instructions for you platform at:

https://www.postgresql.org/download/

-> MySQL: Checking installation
✓ The `mysql` executable was found on your system at: /usr/bin/mysql

-> MySQL: Checking minimum version requirements
✓ Your version of MySQL, 8.0.27, meets the minimum requirements.

-> SQLite3: Checking installation
✓ The `sqlite3` executable was found on your system at: /usr/bin/sqlite3

-> SQLite3: Checking minimum version requirements
✓ Your version of SQLite3, 3.31.1, meets the minimum requirements.

-> Cockroach: Checking installation
✘ The `cockroach` executable could not be found on your system.
For help setting up your Cockroach environment please follow the instructions for you platform at:

https://www.cockroachlabs.com/docs/stable/

-> Buffalo (CLI): Checking installation
✓ The `buffalo` executable was found on your system at: /home/sio4/go/bin/buffalo

-> Buffalo (CLI): Checking minimum version requirements
✓ Your version of Buffalo (CLI), v0.18.0, meets the minimum requirements.

-> Buffalo: Application Details
Pwd         /home/sio4/git/bt/app
Root        /home/sio4/git/bt/app
GoPath      /home/sio4/go
PackagePkg  app
ActionsPkg  app/actions
ModelsPkg   app/models
GriftsPkg   app/grifts
WithModules true
Name        app
Bin         bin/app
VCS         git
WithPop     true
WithSQLite  false
WithDep     false
WithWebpack true
WithNodeJs  true
WithYarn    true
WithDocker  true
WithGrifts  true
AsWeb       true
AsAPI       false
InApp       true
PackageJSON {map[build:webpack --mode production --progress dev:webpack --watch]}

-> Buffalo: config/buffalo-app.toml
name = "app"
bin = "bin/app"
vcs = "git"
with_pop = true
with_sqlite = false
with_dep = false
with_webpack = true
with_nodejs = true
with_yarn = true
with_docker = true
with_grifts = true
as_web = true
as_api = false

-> Buffalo: config/buffalo-plugins.toml
[[plugin]]
  binary = "buffalo-pop"
  go_get = "github.com/gobuffalo/buffalo-pop/v3@latest"

-> Buffalo: go.mod
module app

go 1.16

require (
	github.com/gobuffalo/buffalo v0.18.0
	github.com/gobuffalo/buffalo-pop/v3 v3.0.1
	github.com/gobuffalo/envy v1.10.1
	github.com/gobuffalo/mw-csrf v1.0.0
	github.com/gobuffalo/mw-forcessl v0.0.0-20200131175327-94b2bd771862
	github.com/gobuffalo/mw-i18n/v2 v2.0.1
	github.com/gobuffalo/mw-paramlogger v1.0.0
	github.com/gobuffalo/pop/v6 v6.0.0
	github.com/gobuffalo/suite/v4 v4.0.2
	github.com/markbates/grift v1.5.0
	github.com/unrolled/secure v1.0.9
)
```

</details>

**Comment by sio4** [comment]
#### Current workaround
Just remove `bin/app` or any built binary before running `buffalo build`.

**Comment by sio4** [comment]
Update: will be fixed once v0.18.1 is released.

* #64
